### ISSUE TITLE
Checking for a Sun VM when DCEVM is used.

### ISSUE THREAD

**Issue by electrotype** [electronicarts/ea-agent-loader#6]
I use DCEVM for hot reloading : https://github.com/dcevm/dcevm
When trying to also use `ea-agent-loader`, I get an error because  `String jvm = System.getProperty("java.vm.name").toLowerCase(Locale.ENGLISH);` is equals to "_dynamic code evolution 64-bit server vm_"! It doesn't contain "_hotspot_" or "_openjdk_".

If I patch the `com.ea.agentloader.AgentLoaderHotSpot` class to accept this String as a valid Sun VM, everything works fine.

I'm not sure what would be the best way to fix this.

**Comment by electrotype** [comment]
Note that another workaround, with the current code, is to add `com.sun.tools` as a dependency :

```
<dependency>
    <groupId>com.sun</groupId>
    <artifactId>tools</artifactId>
    <version>${java.version}</version>
    <scope>system</scope>
    <systemPath>${java.home}/../lib/tools.jar</systemPath>
</dependency>
```

**Comment by JoeHegarty** [comment]
Thanks for reporting this.
So just to confirm, the issue is that the check we are running on hotspot fails but if you manually patch the code to accept `"dynamic code evolution 64-bit server vm"` it works correctly?

If that's the case I can add that as an official accepted VM that agent-loader will accept as you've confirmed it works as expected.

The main goal of that check is to make it clear which VMs we support/have been tested.

**Comment by electrotype** [comment]
Yes, this is correct. I added the check to see if the string "dynamic code evolution" was present and everything worked.

But I also said "I'm not sure what would be the best way to fix this." because I don't know if this check would be correct/bulletproof in every situation... Sorry, I can't help more than that.

Thanks.

**Comment by JoeHegarty** [comment]
Can you confirm if the following change functions as you expect?
https://github.com/electronicarts/ea-agent-loader/commit/ff75578e728f399e94e83127ceb390b76a07acb8

If so I can create a release with this.

**Comment by electrotype** [comment]
Yes, this works.

**Comment by JoeHegarty** [comment]
Thanks for your report and thanks for verifying the fix.
This is addressed in v1.0.3 which is now available in Maven Central.

https://repo1.maven.org/maven2/com/ea/agentloader/ea-agent-loader/1.0.3/
https://github.com/electronicarts/ea-agent-loader/releases/tag/v1.0.3

Please let me know if you have any other issues
